doc/specs/#2129 - Side-by-side.md

```diff
@@ -0,0 +1,110 @@
+* Installer Type (MSIX, MSI, EXE)
+* Architecture (x86, x64, ARM, ARM64)
+* Locale (BCP-47 format)
+* Scope (User vs. System)
+
+## Inspiration
+
+Multiple [issues](https://github.com/microsoft/winget-cli/issues?q=is%3Aissue+is%3Aopen+label%3Aside-by-side) have been submitted with the "side-by-side" label.
+
+## Solution Design
+
+Per comment by @JohnMcPMS:
+> Replace PackageVersion InstalledVersion with PackageVersion[] InstalledVersions internally, then reasoning about how that affects everywhere that breaks.
+
+## UI/UX Design
+
+Several different commands will be impacted by this change. In some cases, additional information will be presented to the user so they are aware of the implications for the command executed.
+
+### WinGet Install
+
+When one or more versions (but not the latest) of a package are installed on the system and none of them are pinned, running `winget install <package>` will install the latest version of the package side-by-side unless the manifest specifies the previous version should be uninstalled or denied. The user may override uninstall previous behavior with "--retain-previous", but the deny upgrade behavior would take precedence.
+> [!NOTE]  
+> WinGet will not attempt to override the default behavior the installer may have related to the removal of previous versions of the package.
+
+### WinGet List
+
+When more than one version of a package is installed on the system, all installed versions will be listed. Only the latest version of an installed package will show results in the "Available" column if a newer version is available in a configured source.
+
+### WinGet Upgrade
+
+When more than one version of a package is installed on the system, only the latest installed version will be listed if a newer version is available in a configured source. If the latest available version is in the collection of installed packages, no upgrade will be listed.
+
+### WinGet Uninstall
+
+For `winget uninstall <package>`:
+
+When more than one version of a package is installed on the system, the user will be prompted to disambiguate. The user will be informed "Multiple package versions found matching input criteria. Please specify the version to be uninstalled or use "--all" to remove all versions of the package". All installed versions will be displayed so the user may either disambiguate by specifying the version to be removed or by passing "--all" as an argument to specify every installed version of the package should be uninstalled.
+
+> [!NOTE]
+> The use of "--all" for the uninstall command requires a package to be specified. Attempting to run `winget uninstall --all` is not supported.
+
+For `winget uninstall <package> --version <version>`:
+
+If the version specified is one of the installed versions, only that version will be uninstalled. 
+
+### WinGet Pin
+
+The current behaviors associated with pinned packages will continue to be honored.

```

Comment:
Question on gating pins.

Assume I have the following pin:
```
Name                   Id                       Version Source Pin type
-----------------------------------------------------------------------
Microsoft .NET SDK 3.1 Microsoft.DotNet.SDK_3.1 3.1.42* winget Gating
```

If I have these two versions installed -
```
Name                                                Id                                     Version       Available Source
-------------------------------------------------------------------------------------------------------------------------
Microsoft .NET Core SDK 3.1.426 (x64)               Microsoft.DotNet.SDK.3_1               3.1.426                 winget
Microsoft .NET Core SDK 3.1.426 (x64)               Microsoft.DotNet.SDK.3_1               3.1.456                 winget
```

Would the behavior still be to not upgrade, even though I have 3.1.456 installed? clearly the 3.1.426 shouldn't be upgraded, but would 3.1.456 be omitted from upgrade?